### PR TITLE
check result/errno on all syscalls

### PR DESCRIPTION
**Pull request by ps36094**

### CODE REVIEW

File: client/local.cpp [hunk 1]
```diff
@@ -284,36 +284,45 @@ int build_local(CompileJob &job, MsgChannel *local_daemon, struct rusage *used)
         child_pid = fork();
     }
 
-    if (!child_pid) {
+    if (!child_pid || child_pid == -1) {
```
Comment (HenryMiller1) [inline client/local.cpp]: if `child_pid == -1` we need to log that. 

I'm not sure running the job is correct in the case where we cannot fork.

Comment (ps36094) [inline client/local.cpp]: Ok, I will make that change

File: client/remote.cpp [hunk 2]
```diff
@@ -862,6 +864,11 @@ int build_remote(CompileJob &job, MsgChannel *local_daemon, const Environments &
 
             pid_t pid = fork();
 
+            if (pid == -1) {
+                log_perror("failure of fork");
+                status = -1;
+                }
```
Comment (HenryMiller1) [inline client/remote.cpp]: this brace is indented too far.

File: client/remote.cpp [hunk 3]
```diff
@@ -915,10 +922,14 @@ int build_remote(CompileJob &job, MsgChannel *local_daemon, const Environments &
                         log_error() << umsgs[i]->hostname << " compiled with exit code " << exit_codes[i]
                                     << " and " << umsgs[0]->hostname << " compiled with exit code "
                                     << exit_codes[0] << " - aborting!\n";
-                        ::unlink(jobs[0].outputFile().c_str());
+                        if (-1 == ::unlink(jobs[0].outputFile().c_str())){
+                            log_perror("unlink failed");
```
Comment (HenryMiller1) [inline client/remote.cpp]: We should log the filename here too.

File: daemon/workit.cpp [hunk 4]
```diff
@@ -70,6 +70,8 @@ static int death_pipe[2];
 
 extern "C" {
 
+    int ret; /* added */
```
Comment (HenryMiller1) [inline daemon/workit.cpp]: the commend doesn't seem to add value.

Comment (ps36094) [inline daemon/workit.cpp]: this int ret; has to be removed, and I will do that

**Pull request merged**